packages/daemon/src/app/controller/job.ts

```diff
@@ -17,25 +19,51 @@ export class JobController extends BaseEventController {
   @inject('pluginManager')
   pluginManager: PluginManager;
 
+  private async setupTracer(job: JobEntity) {
+    const logPath = join(constants.PIPCOOK_RUN, job.id, 'logs');
+    const stdoutFile = join(logPath, 'stdout.log');
+    const stderrFile = join(logPath, 'stderr.log');
+    await ensureDir(logPath);
+    await Promise.all([
+      ensureFile(stdoutFile),
+      ensureFile(stderrFile)
+    ]);
+    const tracer = await this.traceManager.create({ stdoutFile, stderrFile });
+    return tracer;
+  }
+
+  private createParams(pipeline: PipelineEntity, jobParams?: JobParam[]): JobParam[] {
+    const params: JobParam[] = [];
+
+    for (const pluginType of constants.PLUGINS) {
+      const tempParam = JSON.parse(pipeline[`${pluginType}Params`]);
```

Comment:
```suggestion
      const defaultParam = JSON.parse(pipeline[`${pluginType}Params`]);
```

Does `defaultParam` or `originalParam` sound better?

Reply:
default should be cool